docs/uploader/xhr.mdx

```diff
@@ -304,6 +247,35 @@ export default {
 
 ## Frequently Asked Questions
 
+### How can I refresh auth tokens after they expire?
+
+```js
+import Uppy from '@uppy/core';
+import XHR from '@uppy/xhr-upload';
+
+let token;
+
+async function getAuthToken(refresh = false) {
+	if (!refresh && token) {
+		return token;
+	}
+	// fetch it
+}
+
+new Uppy().use(XHR, {
+	endpoint: '<your-endpoint>',
+	async onBeforeRequest(xhr) {
+		const token = await getAuthToken();
+		xhr.setRequestHeader('Authorization', `Bearer ${token}`);
+	},
+	async onAfterResponse(xhr) {
+		if (xhr.status === 401) {
+			await getAuthToken(true);
```

Comment:
`onBeforeRequest` is ran, which calls `getAuthToken` and gets the "cached" token. It's not super pretty but it's hard to not do a `fetch` for _every_ file but calling it exactly once when needed (first run and once it expires).

Reply:
ah right i missed the `let token` state since the variable never gets set. It's still unclear to me how this works though. if a request fails with 401, `onAfterResponse` gets called, then `token` gets set to a new token, then how does the original request get re-run with the same parameters as it was the first time?

Reply:
`onBeforeRequest` is called for every retry too and that's where you set the (updated) token.

Reply:
If you can come up with a simpler example how to do this with `onBeforeRequest`, `headers`, `shouldRetry`, `onAfterResponse` I'll happily change it.

Reply:
Done

d44e34a

Reply:
nice